### ISSUE TITLE
How to correctly use the VAE

### ISSUE THREAD

**Issue by hjkl33** [madebyollin/taehv#6]
First of all, thanks for your great work on TAEHV.
When trying to use TAEHV for wan2.1 outside of the demo and integrate it into my own project, I encountered two issues I hope you can help clarify.
1. I noticed that `encode_video` requires the input frame count to be a multiple of 4, whereas the original Wan VAE expects 4n+1 frames. Can the encoder functionality be used properly as expected? Would padding the input resolve this issue?
2. I’ve run the following minimal test code, and the output pixel values range from approximately −0.22 to 1.28, rather than the expected 0-1 range. Is this expected behavior, or am I missing something in the process?
```python
import torch
from taehv import TAEHV

device = "cuda"
dtype   = torch.bfloat16

tiny_vae = TAEHV(
    checkpoint_path="taew2_1.pth",
    decoder_time_upscale=(True, True),
    decoder_space_upscale=(True, True, True)
).to(device, dtype).eval()

# 28 = 7×4 frames, for quick testing
test_x = torch.rand(1, 28, 3, 256, 256, dtype=dtype, device=device)

with torch.no_grad():
    latents = tiny_vae.encode_video(test_x)
with torch.no_grad():
    recon   = tiny_vae.decode_video(latents)
```
I might be missing something obvious, so any guidance you can provide would be invaluable. Thank you once again for your time and for your fantastic work!

**Comment by mshehzadkhan** [comment]
+1

**Comment by madebyollin** [comment]
Thanks for the feedback! I've [pushed](https://github.com/madebyollin/taehv/commit/0b0a47ef98ce3dd8747af9e9cf6e294e342e4ee1) an updated `taehv.py` that should make things less confusing:
1. TAEHV now pads the time axis for you to try and match the reference VAE behaviors. Previously it only matched when the input was a multiple of 4 (and anything non-multiple-of-4 didn't work in parallel mode). Here's the new behavior: <img width="659" height="699" alt="Image" src="https://github.com/user-attachments/assets/f10c5386-db61-4192-8f54-17b0ae8ce6d5" />

2. TAEHV now clamps the decoded output to [0, 1] range for you (even if the network overshoots).

Please try the updated version and LMK any other issues.

**Comment by madebyollin** [comment]
@D1-3105 I added an [example notebook](https://github.com/madebyollin/taehv/blob/main/examples/TAEW2.1_Diffusers_Encoding_and_Decoding_Demo.ipynb) showing how to roundtrip images through all combos of `diffusers.AutoencoderKLWan` &times; `TAEW2_1DiffusersWrapper` - does that help?

<img width="615" height="702" alt="Image" src="https://github.com/user-attachments/assets/ee3f1194-165f-4c2a-9efa-b526eac5ab8b" />

**Comment by xziayro-ai** [comment]
It is now working ! Thx, we were missing the normalization... !
